redash/query_runner/drill.py

```diff
@@ -0,0 +1,239 @@
+import os
+import logging
+import requests
+import re
+
+from dateutil import parser
+
+from redash.query_runner import BaseQueryRunner, register
+from redash.query_runner import TYPE_STRING, TYPE_DATETIME, TYPE_INTEGER, TYPE_FLOAT, TYPE_BOOLEAN
+from redash.utils import json_dumps, json_loads
+
+logger = logging.getLogger(__name__)
+
+
+# Drill returns request result as strings, so we have to guess the actual column type
+def guess_type(string_value):
+    if string_value == '' or string_value is None:
+        return TYPE_STRING
+
+    try:
+        int(string_value)
+        return TYPE_INTEGER
+    except (ValueError, OverflowError):
+        pass
+
+    try:
+        float(string_value)
+        return TYPE_FLOAT
+    except (ValueError, OverflowError):
+        pass
+
+    if unicode(string_value).lower() in ('true', 'false'):
+        return TYPE_BOOLEAN
+
+    try:
+        parser.parse(string_value)
+        return TYPE_DATETIME
+    except (ValueError, OverflowError):
+        pass
+
+    return TYPE_STRING
+
+
+# Convert Drill string value to actual type
+def convert_type(string_value, actual_type):
+    if string_value is None or string_value == '':
+        return ''
+
+    if actual_type == TYPE_INTEGER:
+        return int(string_value)
+
+    if actual_type == TYPE_FLOAT:
+        return float(string_value)
+
+    if actual_type == TYPE_BOOLEAN:
+        return unicode(string_value).lower() == 'true'
+
+    if actual_type == TYPE_DATETIME:
+        return parser.parse(string_value)
+
+    return unicode(string_value)
+
+
+# Parse Drill API response and translate it to accepted format
+def parse_response(data):
+    cols = data['columns']
+    rows = data['rows']
+
+    if len(cols) == 0:
+        return {'columns': [], 'rows': []}
+
+    first_row = rows[0]
+    columns = []
+    types = {}
+
+    for c in cols:
+        columns.append({'name': c, 'type': guess_type(first_row[c]), 'friendly_name': c})
+
+    for col in columns:
+        types[col['name']] = col['type']
+
+    for row in rows:
+        for key, value in row.iteritems():
+            row[key] = convert_type(value, types[key])
+
+    return {'columns': columns, 'rows': rows}
+
+
+class Drill(BaseQueryRunner):
```

Comment:
This should use the `redash.query_runner.BaseHTTPQueryRunner` base class instead since it already implements auth and fetching results via HTTP.